cpp/src/groupby/sort/group_scan_util.cuh

```diff
@@ -78,70 +112,69 @@ struct scan_functor {
     auto result_view = mutable_column_device_view::create(result->mutable_view(), stream);
     auto values_view = column_device_view::create(values, stream);
 
-    if (values.has_nulls()) {
-      auto input = thrust::make_transform_iterator(
-        make_null_replacement_iterator(*values_view, OpType::template identity<DeviceType>()),
-        thrust::identity<ResultDeviceType>{});
+    // Perform segmented scan.
+    auto const do_scan = [&](auto const& inp_iter, auto const& out_iter, auto const& binop) {
       thrust::inclusive_scan_by_key(rmm::exec_policy(stream),
                                     group_labels.begin(),
                                     group_labels.end(),
-                                    input,
-                                    result_view->begin<ResultDeviceType>(),
+                                    inp_iter,
+                                    out_iter,
                                     thrust::equal_to<size_type>{},
-                                    OpType{});
+                                    binop);
+    };
+
+    if (values.has_nulls()) {
+      auto input = thrust::make_transform_iterator(
+        make_null_replacement_iterator(*values_view, OpType::template identity<DeviceType>()),
+        thrust::identity<ResultDeviceType>{});
+      do_scan(input, result_view->begin<ResultDeviceType>(), OpType{});
       result->set_null_mask(cudf::detail::copy_bitmask(values, stream));
     } else {
       auto input = thrust::make_transform_iterator(values_view->begin<DeviceType>(),
                                                    thrust::identity<ResultDeviceType>{});
-      thrust::inclusive_scan_by_key(rmm::exec_policy(stream),
-                                    group_labels.begin(),
-                                    group_labels.end(),
-                                    input,
-                                    result_view->begin<ResultDeviceType>(),
-                                    thrust::equal_to<size_type>{},
-                                    OpType{});
+      do_scan(input, result_view->begin<ResultDeviceType>(), OpType{});
     }
     return result;
   }
+};
 
-  template <typename T>
-  std::enable_if_t<is_supported<T>() and std::is_same_v<T, cudf::string_view>,
-                   std::unique_ptr<column>>
-  operator()(column_view const& values,
-             size_type num_groups,
-             cudf::device_span<cudf::size_type const> group_labels,
-             rmm::cuda_stream_view stream,
-             rmm::mr::device_memory_resource* mr)
+template <aggregation::Kind K, typename T>
+struct group_scan_functor<
+  K,
+  T,
+  std::enable_if_t<is_group_scan_supported<K, T>() and std::is_same_v<T, cudf::string_view>>> {
```

Comment:
See what I said above about not needing to include the logical conjugation of other conditions. We can just do "normal" partial specializations with dispatch to invoke. 
```suggestion
template <aggregation::Kind K>
struct group_scan_functor<
  K,
  cudf::string_view,
  std::enable_if_t<is_group_scan_supported<K, T>()> {
```

Reply:
Yeah, forgot about that---have been using SFINAE too much. Thanks for reminding me.